texts/0005-tauri-config-restructure.md

```diff
@@ -0,0 +1,170 @@
+- Feature Name: tauri_config_restructure
+- Start Date: 2023-13-20
+- RFC PR: [tauri-apps/rfcs#13](https://github.com/tauri-apps/rfcs/pull/13)
+- Tracking Issue: 
+
+# Summary
+
+Restructuring the `tauri.conf.json` for more simplicity, consistency and clarity.
+
+# Motivation
+
+The current `tauri.conf.jsong` has a bunch of objects that don't necessarily follow a rule that could reason why they exist there and thus makes it awkward to figure out where to add new fields
+for example `tauri > bundle` object is a configuration for the CLI, why does it exist inside `tauri` object, another example is the `plugins` object which is a tauri feature, why does it exist outside of `tauri` object? 
+These inconsistencies will make it harder to add new fields to the config. 
+
+# Guide-level explanation
+
+- Unpack `tauri` object to the root object.
+- Unpack `package` fields to the root object.
+- Move `tauri > bundle > identifier` to the root object as it is used by other places in tauri other than bundling.
+- Remove `bundle > updater` as the only useful key it has, is `pubkey` field and that should be moveed to `plugin > updater > pubkey`.
+  This will require changes in the CLI to only sign the bundles when `TAURI_SIGNING_PRIVATE_KEY` key is set .
+  This will also fix a huge DX when using `tauri-plugin-updater` and you endup having to configure updater-related configs once in `bundle > updater` and `plugin > updater`.
+- Move `build > withGlobalTauri` to the root object.
+- Move `tauri > pattern` to `security > pattern` and make it accept a simple string for `brownfield`.
+- Move `tauri > cli` and `tauri > update` fields to `plugins > cli` and `plugins > update` as they are plugins now.
+- Rename `build > distDir` to `build > assets` or just `build > dist`.
```

Comment:
Maybe i'm just too used to distDir but assets sounds too optional to me, like assets on top of your frontend dist (or dev!) files so _i_ would vote for dist or distPath (the latter matching the style of "devUrl" and being a more correct version of the old "distDir" since it can also be an array)

Reply:
the problem with `dist` is people may expect this where the generated executable and installers will be placed. I am not too excited about `distPath` but it has my vote if we can't find a better one.

how about `frontendAssets`, `frontendFiles` or `frontendDist`? too explicit?

Reply:
not too explicit imo. I really like the frontend prefix but i can't decide between Assets/Files/Dist for that 😅

Reply:
`frontendDist` was chosen by the working group poll.